OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/services/ServiceProvider.kt

```diff
@@ -85,13 +85,11 @@ class ServiceProvider(
         }
     }
 
-    // schedule to start all startable services separate threads
+    // schedule to start all startable services in a separate thread
     fun scheduleStartServices() {
-        val provider = this
         coroutineScope.launch {
             synchronized(serviceMap) {
```

Comment:
Since we are now using `getAllServices()`, I believe we can remove this `synchronized`.

Reply:
That's right, synchronized is no longer needed here. I just pushed a change for this.